File: Segundo-Bimestre/Parcial 1/OperacionesMatrices/OperacionesMatrices_python/Reemplazo 0.py
# Matriz donde se reemplace por 0 aquellos valores 
# que se encuentren por encima de la diagonal menor

print("")
n = int(input("Ingrese el tamaño de la matriz: "))

# Declarar la matriz
A = []

# Recorrer la matriz e ingresar los elementos de la matriz
print("\nIngresar los elementos de la matriz")
for i in range (0, n):
    A.append([])
    for j in range (0, n):
        valor = int(input("A["+str(i)+"]["+str(j)+"]: "))
        A[i].append(valor)

print("\nLa matriz A es: ")
for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print("")

print("\nMatriz reemplazada por ceros")
for i in range (0, n):
    for j in range (0, n):
        if (j>i):
            A[i][j] = 0

# Imprimir la matriz reemplazada por 0 a aquellos valores 
# que se encuentren por encima de la diagonal menor

for i in range (0, n):
    for j in range (0, n):
        print(A[i][j], end="   ")
    print("")

File: Segundo-Bimestre/Parcial 1/OperacionesMatrices/OperacionesMatrices_java/src/operaciones/matrices/Reemplazo0.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package operaciones.matrices;
import java.util.Scanner;
/**
 *
 * @author DELL
 */
public class Reemplazo0 {
    public static void main(String[] args) {
        Scanner leer = new Scanner (System.in);
        
        int n, m, i, j;
        n = 0; m = 0; i = 0; j = 0;
        
        //Ingresar el tamaño de la matriz
        System.out.println("---OPERACIONES CON MATRICES---");
        System.out.print("Ingrese el tamaño de la matriz: ");
        n = leer.nextInt();

        
        int[][] A = new int[n][n];
        
        // Recorrer la matriz e ingresar los elementos de la matriz
        System.out.println("\nIngresar los elementos de la matriz");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print("A ["+i+"] ["+j+"] = ");
                A[i][j] = leer.nextInt();
            }
        }
        
        System.out.println("\nLa matriz A es: ");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }

        System.out.println("\nMatriz reemplazada con ceros (valores por encima de la diagonal principal");
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                if (j>i) {
                    A[i][j]=0;
                }
            }
        }        
        
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
                System.out.print(A[i][j]+"   ");
            }
            System.out.println("");
        }
        
    }
}
